src/main/scala/beam/analysis/RideHailFleetAnalysis.scala

```diff
@@ -0,0 +1,393 @@
+    rideHailEvCav.clear()
+    ridehailEvNonCav.clear()
+    rideHailNonEvCav.clear()
+    rideHailNonEvNonCav.clear()
+    processedHour = 0
+  }
+
+  private def assignVehicleDayToLocationMatrix(
+    days: ArrayBuffer[Event],
+    isRH: Boolean,
+    isCAV: Boolean
+  ): (Array[Array[Double]], Array[Array[Double]]) = {
+    val timeUtilization = Array.ofDim[Double](timeBins.size, keys.values.max + 1)
+    val distanceUtilization = Array.ofDim[Double](timeBins.size, keys.values.max + 1)
+    if (isRH) {
+      if (isCAV)
+        timeBins.indices.foreach(timeUtilization(_)(keys("idle")) += 1)
+      else
+        timeBins.indices.foreach(timeUtilization(_)(keys("offline")) += 1)
+    } else {
+      timeBins.indices.foreach(timeUtilization(_)(keys("parked")) += 1)
+    }
+
+    days.zipWithIndex.foreach(eventIndex => {
+      val event = eventIndex._1
+      val idx = eventIndex._2
+      val lastEvent = idx == days.size - 1
+      var chargingNext = false
+      var pickupNext = false
+
+      if (!lastEvent) {
+        val chargingDirectlyNext = days(idx + 1).getEventType == "RefuelSessionEvent"
+
+        val chargingOneAfter =
+          if (idx == days.size - 2)
+            false
+          else
+            days(idx + 1).getEventType == "ParkEvent" && days(idx + 2).getEventType == "RefuelSessionEvent"
+        chargingNext = chargingDirectlyNext || chargingOneAfter
+        pickupNext = days(idx + 1).getEventType == "PathTraversal" && days(idx + 1).getAttributes
+          .get(PathTraversalEvent.ATTRIBUTE_NUM_PASS)
+          .toInt >= 1
+      }
+
+      val eventCharacteristics = classifyEventLocation(event, lastEvent, chargingNext, pickupNext, isRH, isCAV)
+      val eventIdx = keys(eventCharacteristics.eventType)
+
+      val afterDurationEventStart = timeBins
+        .map(timeBin => {
+          val eventStart = timeBin >= eventCharacteristics.start
+          val duringEvent = eventStart && timeBin < eventCharacteristics.end
+          (eventStart, duringEvent)
+        })
+        .unzip
+
+      val afterEventStart = afterDurationEventStart._1
+      val duringEvent = afterDurationEventStart._2
+
+      afterEventStart.zipWithIndex.foreach(indexValue => {
+        if (indexValue._1)
+          timeUtilization(indexValue._2).indices.foreach(timeUtilization(indexValue._2)(_) = 0.0)
+      })
+
+      duringEvent.zipWithIndex.foreach(indexValue => {
+        if (indexValue._1) {
+          timeUtilization(indexValue._2)(eventIdx) += 1.0
+        }
+      })
+
+      if (event.getEventType == "PathTraversal") {
+        val sum = duringEvent.count(during => during)
+        val legLength = event.getAttributes.get(PathTraversalEvent.ATTRIBUTE_LENGTH).toDouble
+        if (sum > 0) {
+          val meanDistancePerTime = legLength / sum
+          duringEvent.zipWithIndex.foreach(indexValue => {
+            if (indexValue._1) {
+              distanceUtilization(indexValue._2)(eventIdx) += meanDistancePerTime / 1609.34
```

Comment:
`1609.34` what's that? Is it possible to extract it to constant?

Reply:
it seems, that it is meters-in-one-mile ..
done